src/Query/ForUpdate.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Query;
+
+/** @internal */
+final class ForUpdate
+{
+    private int $conflictResolutionMode;
+
+    public function __construct(int $conflictResolutionMode)
```

Comment:
This should have `@psalm-param ConflictResolutionMode::* $conflictResolutionMode`, and same for the property type and the return type, as well as other APIs using that pseudo-enum.